rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -0,0 +1,1424 @@
+    def _predict(self, message: Message) -> Optional[Dict[Text, tf.Tensor]]:
+        if self.model is None:
+            logger.error(
+                "There is no trained model: component is either not trained or "
+                "didn't receive enough training data."
+            )
+            return
+
+        # create session data from message and convert it into a batch of 1
+        model_data = self._create_model_data([message])
+
+        return self.model.predict(model_data)
+
+    def _predict_label(
+        self, predict_out: Optional[Dict[Text, tf.Tensor]]
+    ) -> Tuple[Dict[Text, Any], List[Dict[Text, Any]]]:
+        """Predicts the intent of the provided message."""
+
+        label = {"name": None, "confidence": 0.0}
+        label_ranking = []
+
+        if predict_out is None:
+            return label, label_ranking
+
+        message_sim = predict_out["i_scores"].numpy()
+
+        message_sim = message_sim.flatten()  # sim is a matrix
+
+        label_ids = message_sim.argsort()[::-1]
+
+        if (
+            self.component_config[LOSS_TYPE] == "softmax"
+            and self.component_config[RANKING_LENGTH] > 0
+        ):
+            message_sim = train_utils.normalize(
+                message_sim, self.component_config[RANKING_LENGTH]
+            )
+
+        message_sim[::-1].sort()
+        message_sim = message_sim.tolist()
+
+        # if X contains all zeros do not predict some label
+        if label_ids.size > 0:
+            label = {
+                "name": self.inverted_label_dict[label_ids[0]],
+                "confidence": message_sim[0],
+            }
+
+            if (
+                self.component_config[RANKING_LENGTH]
+                and 0 < self.component_config[RANKING_LENGTH] < LABEL_RANKING_LENGTH
+            ):
+                output_length = self.component_config[RANKING_LENGTH]
+            else:
+                output_length = LABEL_RANKING_LENGTH
+
+            ranking = list(zip(list(label_ids), message_sim))
+            ranking = ranking[:output_length]
+            label_ranking = [
+                {"name": self.inverted_label_dict[label_idx], "confidence": score}
+                for label_idx, score in ranking
+            ]
+
+        return label, label_ranking
+
+    def _predict_entities(
+        self, predict_out: Optional[Dict[Text, tf.Tensor]], message: Message
+    ) -> List[Dict]:
+        if predict_out is None:
+            return []
+
+        # load tf graph and session
+        predictions = predict_out["e_ids"].numpy()
+
+        tags = [self.inverted_tag_dict[p] for p in predictions[0]]
+
+        if self.component_config[BILOU_FLAG]:
+            tags = bilou_utils.remove_bilou_prefixes(tags)
+
+        entities = self._convert_tags_to_entities(
+            message.text, message.get("tokens", []), tags
+        )
+
+        extracted = self.add_extractor_name(entities)
+        entities = message.get("entities", []) + extracted
```

Comment:
copy from old code

Reply:
I don't really see this as a reason to not fix it now because these are small things. I guess it's best to collect all these comments in an issue so that we don't forget about them.

Reply:
`CRFEntityExtractor` uses exactly the same code, maybe other extractors as well. Should be fixed everywhere